### PR TITLE
Use unsigned for random seed

### PR DESCRIPTION
**Pull request by howard0su**
Based C++ document and POSIX document, random seed should be an unsigned integer. Change the code to reflect that.

In order to avoid break most the existing applications, let's use -1 as a special value to auto generate a time-based seed still.

Reported in #2002

### CODE REVIEW

File: llama.h [hunk 1]
```diff
@@ -81,7 +81,7 @@ extern "C" {
     typedef void (*llama_progress_callback)(float progress, void *ctx);
 
    struct llama_context_params {
-        int seed;                              // RNG seed, -1 for random
+        unsigned int seed;                     // RNG seed, -1 for random
         int n_ctx;                             // text context
         int n_batch;                           // prompt processing batch size
         int n_gpu_layers;                      // number of layers to store in VRAM
```
Comment (ggerganov) [inline llama.h:87]: Avoid `unsigned int`. Use `uint32_t` and `int32_t`

```suggestion
        uint32_t seed;                             // RNG seed, -1 for random
        int32_t  n_ctx;                            // text context
        int32_t  n_batch;                          // prompt processing batch size
        int32_t  n_gpu_layers;                     // number of layers to store in VRAM
```

File: examples/embedding/embedding.cpp [hunk 2]
```diff
@@ -24,11 +24,11 @@ int main(int argc, char ** argv) {
 
     fprintf(stderr, "%s: build = %d (%s)\n", __func__, BUILD_NUMBER, BUILD_COMMIT);
 
-    if (params.seed < 0) {
+    if (params.seed == -1) {
```
Comment (ggerganov) [inline examples/embedding/embedding.cpp:27]: Wouldn't this cause compiler warnings?
I think better replace `-1` with `0xFFFFFFFF` everywhere, except in the help and the readme

**Review by ggerganov — CHANGES_REQUESTED** [review]

**Comment by axim2** [comment]
Also gpt_params_parse() needs to be changed in examples/common.cpp on line 113: std::stoi => std::stoul

File: llama.cpp [hunk 3]
```diff
@@ -3091,8 +3091,8 @@ int llama_get_kv_cache_token_count(const struct llama_context * ctx) {
 
 #define LLAMA_MAX_RNG_STATE (64*1024)
 
-void llama_set_rng_seed(struct llama_context * ctx, int seed) {
-    if (seed < 0) {
+void llama_set_rng_seed(struct llama_context * ctx, unsigned int seed) {
```
Comment (ggerganov) [inline llama.cpp:3094]:
```suggestion
void llama_set_rng_seed(struct llama_context * ctx, uint32_t seed) {
```

File: llama.h [hunk 4]
```diff
@@ -196,7 +198,7 @@ extern "C" {
     LLAMA_API int llama_get_kv_cache_token_count(const struct llama_context * ctx);
 
     // Sets the current rng seed.
-    LLAMA_API void llama_set_rng_seed(struct llama_context * ctx, int seed);
+    LLAMA_API void llama_set_rng_seed(struct llama_context * ctx, unsigned int seed);
```
Comment (ggerganov) [inline llama.h:201]:
```suggestion
    LLAMA_API void llama_set_rng_seed(struct llama_context * ctx, uint32_t seed);
```

**Review by ggerganov — APPROVED** [review]
Merge after changing `unsigned int` -> `uint32_t`

**Pull request merged**